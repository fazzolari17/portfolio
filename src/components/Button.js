const Button = ({ href, buttonName, style }) => {

  const componentStyle = {
    padding: '1rem 1.5rem',
    fontFamily: 'Lexend Deca, sans-serif',
    maxWidth: 'fit-content',
    backgroundColor: 'var(--hero-background, #395E66)',
    borderRadius: '5px',
    cursor: 'pointer',
    boxShadow: '-1px 0px 38px 8px rgba(0,0,0,0.51)',
  }

  return (
    <div style={{ ...componentStyle, ...style }}>
      <a href={href} target='_blank' rel='noreferrer' >
        {buttonName}
      </a>
    </div>
  )
};

export default Button