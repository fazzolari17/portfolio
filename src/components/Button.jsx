import PropTypes from 'prop-types';
import React from 'react';

const Button = ({ href, buttonName, style, handleClick, invert }) => {

  const styling = invert
    ? { backgroundColor: 'var(--heder-text, #F0F8FF)', color: 'var(--header-background, #071013)' }
    : { backgroundColor: 'var(--header-background, #071013)', color: 'var(--heder-text, #F0F8FF)' };

  const componentStyle = {
    padding: '1rem 1.5rem',
    fontFamily: 'Lexend Deca, sans-serif',
    maxWidth: 'fit-content',
    backgroundColor: 'var(--hero-background, #395E66)',
    borderRadius: '5px',
    cursor: 'pointer',
    boxShadow: '-1px 0px 38px 8px rgba(0,0,0,0.51)',
  };

  return (<div style={{ ...componentStyle, ...style, ...styling }} onClick={handleClick}>
    <a style={styling} href={href} target='_blank' rel='noreferrer' >
      {buttonName}
    </a>
  </div>
  );
};

Button.propTypes = {
  href: PropTypes.string,
  buttonName: PropTypes.string,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  handleClick: PropTypes.func,
  invert: PropTypes.bool
};

export default Button;