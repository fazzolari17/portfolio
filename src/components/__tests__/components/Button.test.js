import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import Button from '../../Button';
import renderer  from 'react-test-renderer';

const mockFn = jest.fn();

describe('button', () => {
  beforeEach(() => {
    render(<Button
      href={'www.testHref.com'}
      buttonName={'test button'}
      handleClick={mockFn}
      invert={false} />);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Button href={'www.testHref.com'}
      buttonName={'test button'}
      handleClick={mockFn}
      invert={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the button with correct href', () => {
    const button = screen.getByRole('link');

    expect(button).toHaveProperty('href', 'http://localhost/www.testHref.com');
  });


  it('renders the button with correct name', () => {
    const buttonName = screen.getByText(/test button/i);

    expect(buttonName).toBeInTheDocument();
  });


  it('calls the function when clicked', () => {
    const button = screen.getByRole(/link/);

    fireEvent.click(button);
    expect(mockFn.mock.calls).toHaveLength(1);

    fireEvent.click(button);
    expect(mockFn.mock.calls).toHaveLength(2);
  });

});