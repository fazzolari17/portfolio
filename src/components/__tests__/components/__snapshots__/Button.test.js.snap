// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders correctly 1`] = `
<div
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "var(--header-background, #071013)",
      "borderRadius": "5px",
      "boxShadow": "-1px 0px 38px 8px rgba(0,0,0,0.51)",
      "color": "var(--header-text, #F0F8FF)",
      "cursor": "pointer",
      "fontFamily": "Lexend Deca, sans-serif",
      "maxWidth": "fit-content",
      "padding": "1rem 1.5rem",
    }
  }
>
  <a
    href="www.testHref.com"
    rel="noreferrer"
    style={
      Object {
        "backgroundColor": "var(--header-background, #071013)",
        "color": "var(--header-text, #F0F8FF)",
      }
    }
    target="_blank"
  >
    test button
  </a>
</div>
`;
